opnode/rollup/derive/payload_attributes.go

```diff
@@ -240,6 +229,14 @@ func ValidBatch(batch *BatchData, config *rollup.Config, epoch rollup.Epoch, min
 	if batch.Timestamp >= maxL2Time {
 		return false // too far in future
 	}
+	for _, txBytes := range batch.Transactions {
+		if len(txBytes) == 0 {
+			return false // transaction data must not be empty
+		}
+		if txBytes[0] == types.DepositTxType {
+			return false // sequencers may not embed any deposits into batch data
+		}
+	}
```

Comment:
Could the sequencer embed an unknown transaction type to halt verifiers? I think that would be fraud provable with cannon but not 100% sure. This extends to sending malformed batches

Reply:
Invalid batches simply get skipped, see https://github.com/ethereum-optimism/optimistic-specs/blob/b7c16710b4463ce79ebbf14f2ad658a04ab004e3/opnode/rollup/derive/payload_attributes.go#L202-L203

So there's no halting of the chain, the sequencer is just wasting their gas on ignored data.

Reply:
The reason why we have to check the deposit type specifically here is because in no case it's valid, and the execution-layer cannot verify the reason for deposit inclusion, only the consensus-layer (rollup node) can during construction of the L2 block inputs.